lib/ukallocbbuddy/bbuddy.c

```diff
@@ -281,6 +331,31 @@ static void *bbuddy_palloc(struct uk_alloc *a, unsigned long num_pages)
 	map_alloc(b, (uintptr_t)alloc_ch, 1UL << order);
 
 	uk_alloc_stats_count_palloc(a, (void *) alloc_ch, num_pages);
+	/* Remove the chunk from the balloon - not for KVM */
```

Comment:
I really don't like having this within the allocator logic in `bbuddy_palloc`. I think that it should be in a separate function especially since there's a lot of duplicate code with other balloon-related stuff in other functions.